core/trino-main/src/test/java/io/trino/execution/scheduler/TestBinPackingNodeAllocator.java

```diff
@@ -380,6 +381,10 @@ public void testNoMatchingNodeAvailableTimeoutReset()
 
             // remove the node - we are again in situation where no matching nodes exist in cluster
             nodeManager.removeNode(NODE_2);
+
+            // sleep for a while before releasing lease, as background processPendingAcquires may be still be running with old snapshot
+            // containing NODE_2, and theNotAcquireLease could be fulfilled when theAcquireLease is released
+            sleepUninterruptibly(10, TimeUnit.MILLISECONDS);
```

Comment:
can this be synchronized without a sleep?

Reply:
Not really - without changing how we setup the test object (that is disabling it's internal thread) - which I think is worse solution as we end up testing something less resembling what we are gonna use in production.
I know it is not optimal but I think it should do the job.